core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/config/RealmConfigXMLProcessor.java

```diff
@@ -181,6 +182,9 @@ private static void addPropertyElements(OMFactory factory, OMElement parent, Str
             Map.Entry<String, String> entry = ite.next();
             String name = entry.getKey();
             String value = entry.getValue();
+            if (value != null) {
```

Comment:
Abstract the null check inside the replaceSystemProperty() method.
Refactor the other two usages of this method accordingly.